irohad/model/converters/impl/pb_query_response_factory.cpp

```diff
@@ -85,6 +85,40 @@ namespace iroha {
         return response;
       }
 
+      optional_ptr<model::QueryResponse> PbQueryResponseFactory::deserialize(
+          const protocol::QueryResponse &response) const {
+        auto p = [](auto &&o) {
+          std::shared_ptr<QueryResponse> result =
+              std::make_shared<std::remove_reference_t<decltype(o)>>(
```

Comment:
I think `decltype(o)` can be extracted as additional variable

Reply:
Not necessary, since it will be replaced with shared model in near future.

Reply:
nevertheless the old code shouldn't be dirty